wagtail/core/tests/test_rich_text.py

```diff
@@ -117,3 +117,61 @@ def test_missing_editor_plugin_returns_none(self):
         self.assertIsNone(
             features.get_editor_plugin('hallo', 'made_up_feature')
         )
+
+
+class TestLinkRewriterTagReplacing(TestCase):
+    def test_should_follow_default_behaviour(self):
+        # we always have default `page` rules registered.
+        rules = {
+            'page': lambda attrs: '<a href="/article/{}">'.format(attrs['id'])
+        }
+        rewriter = LinkRewriter(rules)
+
+        page_type_link = rewriter('<a linktype="page" id="3">')
+        self.assertEqual(page_type_link, '<a href="/article/3">')
+
+        # but it should also be able to handle other supported
+        # link types (email, external) even if no rules is provided
+        external_type_link = rewriter('<a href="https://wagtail.io/">')
+        self.assertEqual(external_type_link, '<a href="https://wagtail.io/">')
+        email_type_link = rewriter('<a href="mailto:test@wagtail.io">')
+        self.assertEqual(email_type_link, '<a href="mailto:test@wagtail.io">')
+
+        # As well as link which don't have any linktypes
+        link_without_linktype = rewriter('<a data-link="https://wagtail.io">')
+        self.assertEqual(link_without_linktype, '<a data-link="https://wagtail.io">')
+
+        # But should not handle if a custom linktype is mentioned but no
+        # associate rules are registered.
+        link_with_custom_linktype = rewriter('<a linktype="custom" href="https://wagtail.io">')
+        self.assertNotEqual(link_with_custom_linktype, '<a href="https://wagtail.io">')
+        self.assertEqual(link_with_custom_linktype, '<a>')
+
+
+    def test_supported_type_should_follow_given_rules(self):
+        # we always have `page` rules by default
+        rules = {
+            'page': lambda attrs: '<a href="/article/{}">'.format(attrs['id']),
+            'external': lambda attrs: '<a target="_blank" href="{}">'.format(attrs['href']),
+            'email': lambda attrs: '<a data-email="true" href="{}">'.format(attrs['href']),
+            'custom': lambda attrs: '<a data-phone="true" href="{}">'.format(attrs['href']),
+        }
+        rewriter = LinkRewriter(rules)
+
+        page_type_link = rewriter('<a linktype="page" id="3">')
+        self.assertEqual(page_type_link, '<a href="/article/3">')
+
+        # It should call appropriate rule supported linktypes (external or email)
+        # based on the href value
+        external_type_link = rewriter('<a href="https://wagtail.io/">')
+        self.assertEqual(external_type_link, '<a target="_blank" href="https://wagtail.io/">')
```

Comment:
Since there is separate checks for `http` and `https`, I think it's worth having tests for both here.